daemon/daemon.go

```diff
@@ -448,6 +448,11 @@ func (daemon *Daemon) registerLink(parent, child *container.Container, alias str
 
 // SetClusterProvider sets a component for querying the current cluster state.
 func (daemon *Daemon) SetClusterProvider(clusterProvider cluster.Provider) {
+	// Daemon is in charge of removing the attachable networks with
+	// connected containers when the node leaves the swarm
+	if clusterProvider == nil {
+		daemon.clearAttachableNetworks()
+	}
```

Comment:
The call from the _cluster_ pkg would have to happen in  `newNodeRunner()` and `Leave()`right before the call to
`c.config.Backend.SetClusterProvider(nil)`

Let's say:
```
c.config.Backend.ClearAttachableNetworks()
c.config.Backend.SetClusterProvider(nil)
```

The fact that they would always need to go in couple and based on the fact that `c==nil` and that the clear attachable networks would really have something to do only when there are user run containers (otherwise it already happen automatically on last task shutdown), which is a notion of the _daemon_, it seem actually ok to me for the call to `clearAttachableNetworks()` to happen inside `SetClusterProvider()`